src/configs/chalk.c42spec

```diff
@@ -453,6 +453,115 @@ failure, and no other keys will be checked.
   }
 }
 
+linguist_types := ["programming", "data", "markup"]
+object linguist_language {
+  user_def_ok:   false
+  gen_fieldname: "linguist_languages"
+  gen_typename:  "LinguistLanguage"
+  gen_setters:   false
+
+  field type {
+    type:    string
+    require: true
+    choice: linguist_types
+  }
+
+  field extension {
+    type:    string
+    require: true
+  }
+
+  field altextensions {
+    type:    list[string]
+    require: false
+  }
+
+}
+
+
+valid_tech_stack_categories := ["database", "webServer", "protocol", "framework"]
+valid_tech_stack_database_types := ["firebird", "hypersonicSQL",
+    "ibmDb2", "microsoftAccess", "microsoftSQLServer",
+     "mongoDB", "mySQL", "oracle", "postgreSQL","sqlite", "sysbase", "other"]
+
+valid_tech_stack_web_server_types := ["apache", "nginx", "iis"]
+valid_tech_stack_protocol_types := ["ldap"]
+valid_tech_stack_framework_types := ["javaSpring"]
+
+# FIXME how can I constraint category w/ subcategory in the tech_stack_rule?
```

Comment:
@viega I will clean those comments up (alongside others in the PR) once its actually opened